backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapper.java

```diff
@@ -105,6 +105,66 @@ private ItemStockSamm listToItemStockSamm(List<? extends ItemStock> itemStocks,
     private record PositionsMappingHelper(String customerOrderId, String supplierOrderId, String customerOrderPositionId) {
     }
 
+    public List<MaterialItemStock> erpSammToMaterialItemStock(ItemStockSamm samm, Partner partner, Material material) {
+        ArrayList<MaterialItemStock> materialItemStocks = new ArrayList<>();
+        for (var position : samm.getPositions()) {
+            String supplierOrderId = null, customerOrderPositionId = null, customerOrderId = null;
+            if (position.getOrderPositionReference() != null) {
+                supplierOrderId = position.getOrderPositionReference().getSupplierOrderId();
+                customerOrderId = position.getOrderPositionReference().getCustomerOrderId();
+                customerOrderPositionId = position.getOrderPositionReference().getCustomerOrderPositionId();
+            }
+            for (var allocatedStock : position.getAllocatedStocks()) {
+                var builder = MaterialItemStock.builder();
+                var itemStock = builder
+                    .partner(partner)
+                    .material(material)
+                    .isBlocked(allocatedStock.getIsBlocked())
+                    .locationBpna(allocatedStock.getStockLocationBPNA())
+                    .locationBpns(allocatedStock.getStockLocationBPNS())
+                    .lastUpdatedOnDateTime(allocatedStock.getLastUpdatedOnDateTime())
+                    .customerOrderId(customerOrderId)
+                    .supplierOrderId(supplierOrderId)
+                    .customerOrderPositionId(customerOrderPositionId)
+                    .measurementUnit(allocatedStock.getQuantityOnAllocatedStock().getUnit())
+                    .quantity(allocatedStock.getQuantityOnAllocatedStock().getValue())
+                    .build();
+                materialItemStocks.add(itemStock);
```

Comment:
`ItemStockErpAdapterService` and `ItemStockSammMapper` currently don't do a validation (beside the implicit validation on the orderPositionRef -> btw we could put this in future in the ItemStockService). During creation there will be an exception. 

As we already have a validation method in the `ItemStockService` subclasses, we should use it either here or in the adapter service and kick them out / don't even call the create (maybe add a log, that amount n invalid stocks have been transferred)

Reply:
We are calling the create methods of MaterialItemStock and ProductItemStock. I.e. the validation there is enforced. The validation methods of the ItemStockServices classes also produces detailed log messages. So the features are already there. 

A different question would perhaps be, if we should reject the entire content of a received reponse message here. I.e.: 
If only one member of the materialItemStockList (or productItemStockList) is rejected by validation, we reject everything from that message (and do not delete all previous stocks for the given partner and material)?

Reply:
Yes, but readability of a stack trace for a business-wise validation is a less comfortable for an admin. You're right, that the more detailed information what's invalid is beneficial.

We should take the information, that is correct. Rejecting the whole message would be too much.